tests/test_misc.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from __future__ import unicode_literals
+
+import pwndbg.commands
+from pwndbg.commands.misc import list_and_filter_commands
+
+STACK_COMMANDS = [
+    ('canary', 'Print out the current stack canary.'),
+    ('context', 'Print out the current register, instruction, and stack context.'),
+    ('down', 'Select and print stack frame called by this one.'),
+    ('retaddr', 'Print out the stack addresses that contain return addresses.'),
+    ('stack', 'dereferences on stack data with specified count and offset.'),
+    ('up', 'Select and print stack frame that called this one.')
+]
+
+
+def test_list_and_filter_commands_filter():
+    assert list_and_filter_commands('stack') == STACK_COMMANDS
```

Comment:
I would prefer that we check contents rather than equality.  I should be able to add a command that mentions `stack` and not break this test.